upload/views/helpers.py

```diff
@@ -32,3 +38,11 @@ def _get_owner_from_string(
         return Owner.objects.get(service=service, username=owner_identifier)
     except Owner.DoesNotExist:
         return None
+
+
+def get_repository_and_owner_from_string(
+    service: Service, repo_identifier: str
+) -> tuple[Repository | None, Owner | None]:
+    return get_repository_from_string(
+        service=service, repo_identifier=repo_identifier, include_owner=True
+    )
```

Comment:
I would actually turn this around:

Move all the logic that does the resolution into `get_repository_and_owner_from_string`, and remove that `include_owner` parameter.
Then call this from `get_repository_from_string`, returning just `result[0]`, which should make things simpler.

The type system can’t reason about that a `True` parameter means you get a `tuple`, and otherwise a `Repository | None`.
So I’m pretty sure the code as is will cause a type error.

Reply:
great suggestion!! updating...